pykafka/cli/kafka_tools.py

```diff
@@ -0,0 +1,285 @@
+import argparse
+import calendar
+import datetime as dt
+
+import tabulate
+
+import pykafka
+from pykafka.common import OffsetType
+
+#
+# Helper Functions
+#
+
+
+def fetch_offsets(client, topic, offset):
+    """Fetch raw offset data from a topic.
+
+    :param client: KafkaClient connected to the cluster.
+    :type client:  :class:`pykafka.KafkaClient`
+    :param topic:  Name of the topic.
+    :type topic:  :class:`pykafka.topic.Topic`
+    :param offset: Offset to reset to. Can be earliest, latest or a datetime.
+        Using a datetime will reset the offset to the latest message published
+        *before* the datetime.
+    :type offset: :class:`pykafka.common.OffsetType` or
+        :class:`datetime.datetime`
+    :returns: dict of {partition_id: :class:`pykafka.protocol.OffsetPartitionResponse`}
+    """
+    if offset.lower() == 'earliest':
+        return topic.earliest_available_offsets()
+    elif offset.lower() == 'latest':
+        return topic.latest_available_offsets()
+    else:
+        offset = dt.datetime.strptime(offset, "%Y-%m-%dT%H:%M:%S")
+        offset = int(calendar.timegm(offset.utctimetuple())*1000)
+        return topic.fetch_offset_limits(offset)
+
+
+def fetch_consumer_lag(client, topic, consumer_group):
+    """Get raw lag data for a topic/consumer group.
+
+    :param client: KafkaClient connected to the cluster.
+    :type client:  :class:`pykafka.KafkaClient`
+    :param topic:  Name of the topic.
+    :type topic:  :class:`pykafka.topic.Topic`
+    :param consumer_group: Name of the consumer group to reset offsets for.
+    :type consumer_groups: :class:`str`
+    :returns: dict of {partition_id: (latest_offset, consumer_offset)}
+    """
+    latest_offsets = fetch_offsets(client, topic, 'latest')
+    consumer = topic.get_simple_consumer(consumer_group=consumer_group,
+                                         auto_start=False)
+    current_offsets = consumer.fetch_offsets()
+    return {p_id: (latest_offsets[p_id].offset[0], res.offset)
+            for p_id, res in current_offsets}
+
+
+#
+# Commands
+#
+
+def desc_topic(client, topic):
```

Comment:
It's really an output function, though. It's just formatting data that's already there for human consumption, hence it being in this tool.